torch/utils/data/datapipes/iter/selecting.py

```diff
@@ -76,7 +76,3 @@ def _isNonEmpty(self, data):
         r = data is not None and \
             not (isinstance(data, list) and len(data) == 0 and self.drop_empty_batches)
         return r
-
-
-    def __len__(self):
```

Comment:
This is being removed because as @ejguan mentioned:

"With `__len__` implemented, `isinstance(dp, Sized)` is always True. I don't think it's the expected behavior."

Reply:
I'm undoing this and moving it to a different PR.